optimize/utils.py

```diff
@@ -0,0 +1,41 @@
+from jax import tree_util, numpy as np
+import numpy as onp
+from typing import Tuple, Callable
+
+
+def flatten_and_unflatten(example_tree) -> Tuple[Callable, Callable]:
```

Comment:
nit: example_tree makes me think this is an example rather than production code

Reply:
Do you have a preferred name for this argument? I chose `example_tree` after rejecting:
* `tree`, since that introduces a name collision with definitions within the function
* `template_tree`, since that suggests the argument is more generic or more template-y, rather than "just an example of what you're trying to flatten"
* `tree_structure`, since that has a specific meaning in jax.tree_util

Perhaps `tree_instance` would be preferred?

Reply:
Perhaps `input_tree` or `base_tree`?

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/474/commits/79586b0b247cdbfbb133a073bc23f817f4222db8